backend/main.py

```
@@ -118,3 +118,49 @@ def read_sales_by_year(year: int, session: SessionDep):
+            status_code=400, detail="Word must contain only English letters"
+        )
+
+    try:
+        db_word = crud.get_word_by_text(session, word_text)
+        if db_word is None:
+            raise HTTPException(status_code=404, detail="Word not found")
+        return db_word
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Failed to get word: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)